Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -605,6 +605,107 @@ - (void)setProxyConfigurations:(NSArray<nw_proxy_config_t> *)proxyConfigurations
     return *_websiteDataStore;
 }
 
+struct WKWebsiteData {
+    std::optional<HashMap<WebCore::ClientOrigin, HashMap<String, String>>> localStorage;
+};
+
+- (void)fetchDataOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(void(^)(NSData *, NSError *))completionHandler
```

Comment:
Please refactor the code so that `fetchDataOfTypes` /  `_fetchDataOfTypes`, and `restoreData` / `_restoreData` will share the same implementation instead of having duplicate code.  E.g. adding a `fetchDataOfTypesShared` function.